torch/csrc/PyInterpreter.cpp

```diff
@@ -663,8 +663,7 @@ c10::SymIntArrayRef ConcretePyInterpreterVTable::sym_sizes(
   for (auto it = out.begin(); it != out.end(); it++) {
     auto elm = *it;
     auto si = py::cast<c10::SymInt>(elm);
-    // TODO: the buffer will need to be made owning later
-    symints.append(si.as_int_unchecked());
+    symints.append(si);
```

Comment:
Is there any special handling that needs to be done here further down in `values_from_buffer` to handle SymInts? Currently it just does `THPUtils_unpackLongs`.

Reply:
whoops, thanks. My PR removes the caching logic anyway, so I'm tentatively just going to delete the `values_from_buffer` code entirely (no reason to stash anything in a python buffer if we aren't caching it).

Reply:
You can't remove values_from_buffer, it is needed to ensure the returned IntArrayRef stays live beyond the extent of the size() call